Add new docs (#287)
learning-process/parallel_programming_course: docs/common_information/introduction.rst

Introduction
============

Practice:
~~~~~~~~~

- We work online

  - Use GitHub repository sysytem
  - Use Pull Requests
  - Merge into the master branch
  - Test verification

> - Task distribution is random for each person.

Reviewer: ```suggestion
- Task distribution is random for each student.
```

> - An example for each technology can be found in the directory:

Reviewer: ```suggestion
- An example for each technology can be found in the corresponding directory:
```

  ``tasks/<all/omp/tbb/seq/stl/mpi>/example``.

> - In each repository, the README.md contains a link to the build
>   instructions (**read it in full!!!**).

Reviewer: ```suggestion
- In each repository, the README.md contains a link to the course
  documentation (**read it fully!!!**).
```

- Additionally, each repository includes an example of a properly
  formatted PULL REQUEST.

> - All tasks are mandatory.

Reviewer: ```suggestion
- Submission of all tasks is mandatory to pass the course.
```

- A task that has been merged into the master branch continues to be
  reviewed. If a task fails in the master, it is disabled, and a record
  of this is added to the score table. All disabled tasks will result in

>   a zero score for those tasks at the end of the semester.

Reviewer: ```suggestion
  a zero points result for those tasks at the end of the semester.
```

- All resources for using the repository will be provided here:

>   - `GIT-1 <https://proglib.io/p/git-for-half-an-hour>`__
>   - `GIT-2 <https://medium.com/nuances-of-programming/%D0%B7%D0%BD%D0%B0%D0%BA%D0%BE%D0%BC%D1%81%D1%82%D0%B2%D0%BE-%D1%81-git-%D0%B8-github-%D1%80%D1%83%D0%BA%D0%BE%D0%B2%D0%BE%D0%B4%D1%81%D1%82%D0%B2%D0%BE-%D0%B4%D0%BB%D1%8F-%D0%BD%D0%B0%D1%87%D0%B8%D0%BD%D0%B0%D1%8E%D1%89%D0%B8%D1%85-54ea2567d76c>`__
>   - `GIT-3 <https://tproger.ru/translations/git-quick-start/>`__
>   - `GIT CONFILCTS <https://githowto.com/ru/resolving_conflicts>`__
>   - `GTEST-1 <https://habr.com/ru/post/119090/>`__
>   - `GTEST-2 <https://github.com/google/googletest/blob/master/googletest/docs/primer.md>`__
>   - `GITHUB ACTIONS <https://docs.github.com/ru/actions>`__
>   - `MPI <https://parallel.ru/vvv/mpi.html#p1>`__
>   - `LaTeX-1 <https://www.mccme.ru/free-books/llang/newllang.pdf>`__
>   - `LaTeX-2 <http://www.docs.is.ed.ac.uk/skills/documents/3722/3722-2014.pdf>`__
>   - `OpenMP <https://parallel.ru/tech/tech_dev/openmp.html>`__
>   - `TBB-1 <http://www.unn.ru/pages/e-library/aids/2007/12.pdf>`__
>   - `TBB-2 <https://software.intel.com/content/www/ru/ru/develop/articles/tbb_async_io.html>`__
>   - `std::thread-1 <https://eax.me/cpp-multithreading/>`__
>   - `std::thread-2 <http://www.quizful.net/post/multithreading-cpp11>`__
>   - `std::thread-3 <http://ikt.muctr.ru/images/info/2Lecture_3.pdf>`__

Reviewer: ```suggestion
  - `Git for half an hour: A Beginner’s Guide <https://proglib.io/p/git-for-half-an-hour>`__
  - `Getting Started with Git and GitHub: A Beginner’s Guide <https://medium.com/nuances-of-programming/%D0%B7%D0%BD%D0%B0%D0%BA%D0%BE%D0%BC%D1%81%D1%82%D0%B2%D0%BE-%D1%81-git-%D0%B8-github-%D1%80%D1%83%D0%BA%D0%BE%D0%B2%D0%BE%D0%B4%D1%81%D1%82%D0%B2%D0%BE-%D0%B4%D0%BB%D1%8F-%D0%BD%D0%B0%D1%87%D0%B8%D0%BD%D0%B0%D1%8E%D1%89%D0%B8%D1%85-54ea2567d76c>`__
  - `Git: A Quick Start Guide to Using Core Operations with Explanations <https://tproger.ru/translations/git-quick-start/>`__
  - `Conflicts resolving in Git <https://githowto.com/ru/resolving_conflicts>`__
  - `Google testing framework (gtest) <https://habr.com/ru/post/119090/>`__
  - `GoogleTest Primer <https://github.com/google/googletest/blob/main/docs/primer.md>`__
  - `GitHub Actions documentation <https://docs.github.com/ru/actions>`__
  - `Parallel Programming Technologies. Message Passing Interface (MPI) <https://parallel.ru/vvv/mpi.html#p1>`__
  - `Typing and Layout in the System LaTeX <https://www.mccme.ru/free-books/llang/newllang.pdf>`__
  - `LaTeX for the beginners <https://web.archive.org/web/20210420184700/http://www.docs.is.ed.ac.uk/skills/documents/3722/3722-2014.pdf>`__
  - `What is OpenMP? <https://parallel.ru/tech/tech_dev/openmp.html>`__
  - `TBB-1 <https://web.archive.org/web/20220406024622/http://www.unn.ru/pages/e-library/aids/2007/12.pdf>`__
  - `Writing Multithreaded Applications in C++ <https://eax.me/cpp-multithreading/>`__
  - `Multithreading: New Features of the C++11 Standard <https://web.archive.org/web/20200608173050/http://www.quizful.net/post/multithreading-cpp11>`__
  - `Introduction to Parallel Computing <http://ikt.muctr.ru/images/info/2Lecture_3.pdf>`__
```

\* *All instructions, repositories, and tables may be updated during the

> learning process for better usability. Be prepared for changes and check

Reviewer: ```suggestion
learning process for better usability. Be prepared for changes, check and update
```
